cmake/Modules/MacroQbtCXXCompilerAndModeCheck.cmake

```diff
@@ -0,0 +1,47 @@
+macro(qbt_cxx_compiler_and_mode_check)
+    # some compilers advertise support for the minimum required version (currently 14),
+    # but it may not be full support
+    # Refer to the C++ mode support matrices on the websites of each compiler
+    set(QBITTORRENT_MINIMUM_CXX_STANDARD "14")
+    set(QBITTORRENT_MINIMUM_GCC_VERSION "5.0")
+    set(QBITTORRENT_MINIMUM_CLANG_VERSION "3.4")
+    set(QBITTORRENT_MINIMUM_MSVC_VERSION "1910")
+    if ("cxx_std_${QBITTORRENT_MINIMUM_CXX_STANDARD}" IN_LIST CMAKE_CXX_COMPILE_FEATURES)
+        if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+            if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "${QBITTORRENT_MINIMUM_GCC_VERSION}")
+                message(FATAL_ERROR "qBittorrent requires GCC >= ${QBITTORRENT_MINIMUM_GCC_VERSION}\n"
+                "for full C++${QBITTORRENT_MINIMUM_CXX_STANDARD} support")
+            endif()
+        elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+            if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "${QBITTORRENT_MINIMUM_CLANG_VERSION}")
+                message(FATAL_ERROR "qBittorrent requires Clang >= ${QBITTORRENT_MINIMUM_CLANG_VERSION}\n"
+                "for full C++${QBITTORRENT_MINIMUM_CXX_STANDARD} support")
+            endif()
+        elseif(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+            if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "${QBITTORRENT_MINIMUM_MSVC_VERSION}")
+                message(FATAL_ERROR "qBittorrent requires MSVC >= ${QBITTORRENT_MINIMUM_MSVC_VERSION}\n"
+                "for full C++${QBITTORRENT_MINIMUM_CXX_STANDARD} support")
+            endif()
+        else()
+            message(WARNING "Using untested compiler/compiler version\n"
+            "build might fail due to incomplete C++${QBITTORRENT_MINIMUM_CXX_STANDARD} support or other reasons.")
+        endif()
+    else()
+        message(FATAL_ERROR "qBittorrent requires a compiler with full C++${QBITTORRENT_MINIMUM_CXX_STANDARD} support,\n"
+                "such as GCC >= ${QBITTORRENT_MINIMUM_GCC_VERSION}\n"
+                "or Clang >= ${QBITTORRENT_MINIMUM_CLANG_VERSION}")
+    endif()
+
+    # now we know that the compiler fully supports the minimum required C++ mode,
+    # but we must still prevent the user or compiler from forcing/defaulting to an insufficient C++ mode
```

Comment:
Just set `PUBLIC` target feature for the base target. See https://github.com/arvidn/libtorrent/blob/RC_1_2/cmake/Modules/LibtorrentMacros.cmake#L66

Reply:
What's the advantage? And wouldn't I have to do that in the `CMakeLists.txt` under `src/`?

Reply:
Your code comment says you want to prevent user to override the standard value, which is very much understandable given the libtorrent standard-dependent ABI. Users can't override target features.

Reply:
@zeule 
> Your code comment says you want to prevent user from overriding the standard value,

No, I just want to prevent the user from using a lower than minimum value. For example, if the default value is `14`, the user is allowed to set `17` but not `11`.